editor.planx.uk/src/lib/yupExtensions.test.ts

```diff
@@ -0,0 +1,167 @@
+import { richText } from "./yupExtensions";
+
+describe("richText validation", () => {
+  describe("link formatting", () => {
+    it("does not display an error if the text '(opens in a new tab)' is wrapped in an anchor element", () => {
+      const result = richText().validate(
+        '<a href="#">Some link (opens in a new tab)</a>',
+      );
+      expect(result).toBeDefined();
+    });
+
+    it("displays an error if the text '(opens in a new tab)' is not wrapped in an anchor element", async () => {
+      await expect(() =>
+        richText().validate(
+          '<p><a href="#">Some link</a> (opens in a new tab)</p>',
+        ),
+      ).rejects.toThrow('Links must wrap the text "(opens in a new tab)".');
+    });
+
+    it("displays an error if the policy legislation link contains '/made'", async () => {
+      await expect(() =>
+        richText().validate(
+          '<p><a href="https://www.legislation.gov.uk/ukpga/2023/36/section/3/made">Link to policy</a></p>',
+        ),
+      ).rejects.toThrow(
+        'Legislative policy links should not end in "/made" as these can be out of date.',
+      );
+    });
+
+    it("does not display an error if the policy legislation link does not contain '/made'", () => {
+      const result = richText().validate(
+        '<p><a href="https://www.legislation.gov.uk/ukpga/2023/36/section/3">Link to policy</a></p>',
+      );
+      expect(result).toBeDefined();
+    });
+
+    it("displays an error for short links (fewer than 8 characters)", async () => {
+      await expect(() =>
+        richText().validate('<p><a href="https://www.planx.uk">Short!</a></p>'),
+      ).rejects.toThrow(
+        "Make sure the link text accurately describes the what the link is for.",
+      );
+    });
+
+    it("displays an error for not non-descriptive links", async () => {
+      await expect(() =>
+        richText().validate(
+          '<p><a href="https://www.planx.uk">Click here!</a></p>',
+        ),
+      ).rejects.toThrow(
+        "Links must be set over text that accurately describes what the link is for. Avoid generic language such as 'click here'.",
+      );
+
+      await expect(() =>
+        richText().validate(
+          '<p><a href="https://www.planx.uk">Try clicking here!</a></p>',
+        ),
+      ).rejects.toThrow(
+        "Links must be set over text that accurately describes what the link is for. Avoid generic language such as 'click here'.",
+      );
+    });
+  });
+
+  describe("content hierarchy (headings)", () => {
```

Comment:
@ianjon3s Would appreciate an eye on these tests - do they correctly cover the behaviour we expect to see?

One issue I've spotted here is that for `rootLevelContent`, we can still have `h1+h3+h2` which is incorrect but not covered by the current implementation.